src/store/Controller.cc

```diff
@@ -837,22 +809,41 @@ Store::Controller::anchorToCache(StoreEntry &entry, bool &inSync)
 
     debugs(20, 7, "anchoring " << entry);
 
+    Transients::EntryStatus entryStatus;
+    transients->status(entry, entryStatus);
+
+    inSync = false;
     bool found = false;
     if (sharedMemStore)
         found = sharedMemStore->anchorToCache(entry, inSync);
     if (!found && swapDir)
         found = swapDir->anchorToCache(entry, inSync);
 
+    if (inSync) {
+        debugs(20, 7, "anchored " << entry);
+        assert(found);
+        entry.setCollapsingRequirement(false);
+        return true;
+    }
+
     if (found) {
-        if (inSync)
-            debugs(20, 7, "anchored " << entry);
-        else
-            debugs(20, 5, "failed to anchor " << entry);
-    } else {
-        debugs(20, 7, "skipping not yet cached " << entry);
+        debugs(20, 5, "failed to anchor " << entry);
+        return true;
+    }
+
+    if (entryStatus.waitingToBeFreed) {
+        debugs(20, 5, "failed on marked unattached " << entry);
+        return true;
```

Comment:
> since 'found' is now true when waitingToBeFreed is true.

Sorry, I do not follow. These changes do not change the values of `found` and `waitingToBeFreed` IIIRC. Also, there is no relationship between the two: `found` is about finding the matching entry in one of the stores while the `waitingToBeFreed` flag is in the transients entry itself. All four combinations of values are possible. What am I missing?